scripts/dataset-fetch.py

```diff
@@ -38,12 +38,19 @@ def wget(cfg, batch, index):
     print "Fetching batch %d" % index
     urls = [cfg.url + chunk for chunk in batch]
     pchunk = int(math.ceil(len(urls)/ncpus))
-    pargs = (urls[x:x+pchunk] for x in range(0, len(urls), pchunk))
+    pargs = [urls[x:x+pchunk] for x in range(0, len(urls), pchunk)]
     procs = [wget1(cfg, args) for args in pargs]
-    for p in procs:
+    retry = []
+    for x, p in enumerate(procs):
         rc = p.wait()
         if rc != 0:
-            raise Exception("Error fetching data: wget exit code %d" % rc)
+            retry += pargs[x]
```

Comment:
I finally got the script work reliably, so I am not inclined to change it. 
I'll keep parallel in mind, and maybe use it next time a similar task arises.

Re: progress reporting: is just spam in the console, which occludes the progress and timings from the larger ingestion script.